astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -565,9 +603,35 @@ def divide(self, operand, operand2=None, **kwargs):
             np.true_divide, operand, operand2, **kwargs
         )
 
+    @sharedmethod
+    def sum(self, **kwargs):
+        return self._prepare_then_do_arithmetic(np.sum, **kwargs)
+
+    @sharedmethod
+    def mean(self, **kwargs):
+        return self._prepare_then_do_arithmetic(np.mean, **kwargs)
+
+    @sharedmethod
+    def min(self, **kwargs):
+        return self._prepare_then_do_arithmetic(np.min, **kwargs)
+
+    @sharedmethod
+    def max(self, **kwargs):
+        return self._prepare_then_do_arithmetic(np.max, **kwargs)
+
+    @sharedmethod
+    def amin(self, **kwargs):
```

Comment:
Is it necessary to maintain a redundant API from another package?  (numpy in this case).  My understanding is that `NDDataArray` doesn't require that the data array be a numpy array.  It could be another type of array.  In that case, this API would make no sense.  Would it be better to stick to the philosophy that there should be only one obvious way of doing something?  And hence only support `min`/`max` and not `amin`/`amax`?